enhancements/monitoring/integration-with-topology-modes.md

```diff
@@ -0,0 +1,90 @@
+---
+title: monitoring-topology-mode
+authors:
+  - "@lilic"
+reviewers:
+  - "monitoring-team"
+  - "???"
+approvers:
+  - "monitoring-team"
+creation-date: 2021-03-01
+last-updated: 2021-03-01
+status: implementable
+see-also:
+  - "/enhancements/cluster-high-availability-mode-api.md"
+---
+
+# Monitoring Stack on Single Replica Topology Mode
+
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Monitoring stack needs to react based on the newly added single replica topology mode. Plan is to detect whenever
+cluster is in single replica topology mode and reduce the monitoring stack footprint by adjusting platform Prometheus
+instance to be non Highly Available.
+
+## Motivation
+
+In the 4.8 time frame, OpenShift will introduce a new capabilities field in the Infrastructure API that will inform
```

Comment:
Sounds good, we only care about controlPlaneTopology for Prometheus. Should I say its a non goal we do anything in infraTopology fields?

Reply:
My understanding is that `controlPlaneTopology` is specifically for components that run on control nodes. Since the monitoring stack (except CMO which is already single replica) is deployed on worker nodes, we should rely on `infrastructureTopology`.
@dhellmann is this correct?

Reply:
yes

Reply:
`infrastructureTopology` does sound like the right field to observe.

How does the monitoring stack work in the HyperShift case, where the control plane is outside of the cluster but there may be "infrastructure" nodes in the cluster?